scripts/new_project.kts

```diff
@@ -59,38 +67,60 @@ object NewProject {
     }
 
     private fun handleArguments(args: Array<String>) {
-        val argumentError = "ERROR: Invalid argument name: Ensure define arguments => app-name={\"MyProject\"} or {\"My Project\"} package-name={com.sample.myproject}"
+        val argumentError = "ERROR: Invalid argument name: $HELP_MESSAGE"
         when (args.size) {
             1 -> when {
-                args.first().startsWith(KEY_APP_NAME) -> showErrorMessage("ERROR: No package name has been provided")
-                args.first().startsWith(KEY_PACKAGE_NAME) -> showErrorMessage("ERROR: No app name has been provided")
-                else -> showErrorMessage(argumentError)
+                args.first() == KEY_HELP -> showMessage(
+                    message = HELP_MESSAGE,
+                    exitAfterMessage = true
+                )
+                args.first().startsWith(KEY_APP_NAME) -> showMessage(
+                    message = "ERROR: No package name has been provided",
+                    exitAfterMessage = true
+                )
+                args.first().startsWith(KEY_PACKAGE_NAME) -> showMessage(
+                    message = "ERROR: No app name has been provided",
+                    exitAfterMessage = true
+                )
+                else -> showMessage(
+                    message = argumentError,
+                    exitAfterMessage = true
+                )
             }
             2 -> args.forEach { arg ->
                 val (key, value) = arg.split(ARGUMENT_DELIMITER)
                 when (key) {
                     KEY_APP_NAME -> validateAppName(value)
                     KEY_PACKAGE_NAME -> validatePackageName(value)
-                    else -> showErrorMessage(argumentError)
+                    else -> showMessage(message = argumentError, exitAfterMessage = true)
                 }
             }
-            else -> showErrorMessage("ERROR: Require app-name and package-name to initialize the new project")
+            else -> showMessage(
```

Comment:
Not sure if we should update the logic or the error message here 🤔 
When entering `kscript new_project.kts package-name="my.package" app-name="MyApp" --help`, the error says `Error: Require app-name and package-name...` even though all three arguments are valid 🤔

Reply:
@ryan-conway Good point 👍 , I think we just show the `HELP_MESSAGE` in case of `else`, Same for line `85` and `95`

Reply:
I think we also miss handle this case

<img width="1220" alt="Screen Shot 2565-08-04 at 11 33 52" src="https://user-images.githubusercontent.com/13492460/182764044-2b9c44f6-4e38-438b-864e-13ed909f2c00.png">

Reply:
Look like we shouldn't base the logic on argument count anymore, and instead should check to see if all the required arguments are there or if an invalid argument exists 🤔 How about something like this?
```kotlin
        var hasPackage = false
        var hasName = false
        args.forEach { arg ->
            val (key, value) = arg.split(ARGUMENT_DELIMITER)
            when (key) {
                KEY_APP_NAME -> {
                    validateAppName(value)
                    hasName = true
                }
                KEY_PACKAGE_NAME -> {
                    validatePackageName(value)
                    hasPackage = true
                }
                KEY_HELP -> showMessage(
                    message = HELP_MESSAGE,
                    exitAfterMessage = true
                )
                else -> showMessage(
                    message = "ERROR: Invalid argument name: $key $HELP_MESSAGE",
                    exitAfterMessage = true
                )
            }
        }
        when {
            !hasName -> showMessage(
                message = "ERROR: No app name has been provided",
                exitAfterMessage = true
            )
            !hasPackage -> showMessage(
                message = "ERROR: No package name has been provided",
                exitAfterMessage = true
            )
        }
```

Reply:
What do you think if we display a help message with a generic argument error on every error case? I think it should be enough, for instance

```
        val argumentError = "ERROR: Invalid argument: $HELP_MESSAGE"
        if (args.size == 1 && args.first() == KEY_HELP) {
            showMessage(message = HELP_MESSAGE, exitAfterMessage = true)
        } else if (args.size == 2) {
            args.forEach { arg ->
                try {
                    val (key, value) = arg.split(ARGUMENT_DELIMITER)
                    when (key) {
                        KEY_APP_NAME -> validateAppName(value)
                        KEY_PACKAGE_NAME -> validatePackageName(value)
                        else -> showMessage(message = argumentError, exitAfterMessage = true)
                    }
                } catch (exception: Exception) {
                    showMessage(message = argumentError, exitAfterMessage = true)
                }
            }
        } else {
            showMessage(message = argumentError, exitAfterMessage = true)
        }
```

@kaungkhantsoe @Wadeewee @ryan-conway @Tuubz

Reply:
@chornerman I really don't think the arg size check is the right call, because if we run `kscript new_project.kts package-name="my.package" package-name="my.package.again"` there's still two valid args even though `app-name` wasn't set 🤔

Reply:
Yeah, I agreed, @chornerman. As I recommended in this comment https://github.com/nimblehq/android-templates/pull/260#discussion_r934337402 to make it simple, but I think you need to update `logic` a bit. Let's use `when` instead of `if/else`

Reply:
@ryan-conway From your suggestion, I think if we run something like `kscript new_project.kts --help` it will crash since we can't split `--help`. How about if we combine our idea together 😄 

```
        var hasPackage = false
        var hasName = false
        val argumentError = "ERROR: Invalid argument\n$HELP_MESSAGE"
        if (args.size == 1 && args.first() == KEY_HELP) {
            showMessage(message = HELP_MESSAGE, exitAfterMessage = true)
        } else if (args.size == 2) {
            var hasPackage = false
            var hasName = false
            args.forEach { arg ->
                try {
                    val (key, value) = arg.split(ARGUMENT_DELIMITER)
                    when (key) {
                        KEY_APP_NAME -> {
                            validateAppName(value)
                            hasName = true
                        }
                        KEY_PACKAGE_NAME -> {
                            validatePackageName(value)
                            hasPackage = true
                        }
                        else -> showMessage(message = argumentError, exitAfterMessage = true)
                    }
                } catch (exception: Exception) {
                    showMessage(message = argumentError, exitAfterMessage = true)
                }
            }
            if (!hasName || !hasPackage) showMessage(message = argumentError, exitAfterMessage = true)
        } else {
            showMessage(message = argumentError, exitAfterMessage = true)
        }
```

Reply:
@chornerman Works for me 😄

Reply:
@chornerman @ryan-conway I think we can simply check if `args.size > 1` then check further for the argument key. If we use `==2`, then in the future if we add more argument, we need to update the condition too.
```
var hasValidPackage = false
        var hasValidName = false
        val argumentError = "ERROR: Invalid argument\n$HELP_MESSAGE"
        when {
            args.size == 1 && args.first() == KEY_HELP -> {
                showMessage(message = HELP_MESSAGE, exitAfterMessage = true)
            }
            args.size > 1 -> {
                args.forEach { arg ->
                    try {
                        val (key, value) = arg.split(ARGUMENT_DELIMITER)
                        when (key) {
                            KEY_APP_NAME -> {
                                validateAppName(value)
                                hasValidName = true
                            }
                            KEY_PACKAGE_NAME -> {
                                validatePackageName(value)
                                hasValidPackage = true
                            }
                            else -> showMessage(message = argumentError, exitAfterMessage = true)
                        }
                    } catch (exception: Exception) {
                        showMessage(message = argumentError, exitAfterMessage = true)
                    }
                }
                if (!hasValidName || !hasValidPackage) showMessage(message = argumentError, exitAfterMessage = true)
            }
            else -> {
                showMessage(message = argumentError, exitAfterMessage = true)
            }
        }
```

Reply:
I've adapted @ryan-conway's suggestion a bit.

It would definitely be better if we can move away from checking the argument count:
```
  private fun handleArguments(args: Array<String>) {
        var hasAppName = false
        var hasPackageName = false
        args.forEach { arg ->
            if (arg == KEY_HELP) {
                showMessage(
                    message = HELP_MESSAGE,
                    exitAfterMessage = true
                )
            } else if (arg.contains("$KEY_APP_NAME$ARGUMENT_DELIMITER")) {
                val (key, value) = arg.split(ARGUMENT_DELIMITER)
                validateAppName(value)
                hasAppName = true
            } else if (arg.contains("$KEY_PACKAGE_NAME$ARGUMENT_DELIMITER")) {
                val (key, value) = arg.split(ARGUMENT_DELIMITER)
                validatePackageName(value)
                hasPackageName = true
            } else {
                showMessage(
                    message = "ERROR: Invalid argument name: $arg \n$HELP_MESSAGE",
                    exitAfterMessage = true
                )
            }
        }
        when {
            !hasAppName -> showMessage(
                message = "ERROR: No app name has been provided \n$HELP_MESSAGE",
                exitAfterMessage = true
            )
            !hasPackageName -> showMessage(
                message = "ERROR: No package name has been provided \n$HELP_MESSAGE",
                exitAfterMessage = true
            )
        }
    }
```

Hopefully, I didn't miss any cases 🙏

Reply:
@Tuubz Shall we use `arg.startsWith()` instead of `arg.contains()`? Just to prevent inputs like `some-app-name=` 🤔

Reply:
@Tuubz @ryan-conway Look neat! How about if we also show the `helpMessage` when displaying a validate app name and package name error? 😄

Reply:
One more thing, should we also add the `{my-project}` pattern to the validate app name error message to match the help message? 🤔 Something like
`ERROR: Invalid App Name: $value (needs to follow standard pattern {MyProject} or {My Project} or {my-project})`